blivet/devicetree.py

```diff
@@ -165,12 +171,6 @@ def _add_device(self, newdev, new=True):
         newdev.add_hook(new=new)
         self._devices.append(newdev)
 
-        # don't include "req%d" partition names
-        if ((newdev.type != "partition" or
-             not newdev.name.startswith("req")) and
-                newdev.type != "btrfs volume" and
-                newdev.name not in self.names):
-            self.names.append(newdev.name)
```

Comment:
It looks like the property drops these special cases (unallocated partitions w/ names like 'req1', btrfs volumes w/ same name as first backing device). It should be fairly simple to preserve at least the partition one.

Reply:
Fixed. Instead of removing, the code was moved into property handling function.